api/candlepin-api-spec.yaml

```diff
@@ -515,6 +515,17 @@ components:
               additionalProperties:
                 type: string
 
+    ArrayExcludeGuestIdDTO:
```

Comment:
The name seems odd to me, and probably more references the original HATEOAS usage rather than what the DTO itself *is*. I feel that something like AttributelessGuestIdDTO would be clearer going forward; assuming we also have a GuestIdDTO that *does* have the attributes associated with it.

Reply:
Yes we do have GuestIdDTO that does have the attributes, already in the spec file. The reason I just used the original hateoas usage was because we'll have a couple more DTOs that will need to have cut-down versions because they use that annotation (specifically, ConsumerDTO and UpstreamConsumerDTO, which I've updated the relevant jira cards to make sure they get created), and wanted to use a standard naming scheme for all. For example, ConsumerDTO has that annotation for the `idCert` and `facts` fields, and naming it `IdCertLessAndFactLessConsumerDTO` is not a good idea.

Maybe a better format for all 3 of them would be `ArrayElement*DTO` or `*DTOArrayElement`, because those DTOs are only going to be used as references in an array?

Reply:
The *DTOArrayElement for standardization sounds good to me.

Reply:
Done